### PR TITLE
[CORE] [CLANG] Fix warnings reported by llvm16 in CLANG IBs

### PR DESCRIPTION
**Pull request by smuzaffar**
- Fixed set but unused warnings from clang-16 in CLANG IBs.
- Added couple of cout sothat compiler do not complain about set but unused variables.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+core

**Comment by perrotta** [comment]
+1

File: DataFormats/Common/test/testOneToManyAssociation.cc [hunk 1]
```diff
@@ -67,6 +68,7 @@ void testOneToManyAssociation::dummy() {
     ++f;
     n = v.numberOfAssociations(edm::Ref<CKey>());
     ++n;
+    std::cout << "numberOfAssociations:" << n << std::endl;
```
[the remark this answers is not in the document]
Comment (smuzaffar) [reply DataFormats/Common/test/testOneToManyAssociation.cc:71]: @makortel , I am not sure if unit test is testing this code, I do not see this printout in the log file https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-07754a/32373/unitTests/src/DataFormats/Common/test/testDataFormatsCommon/testing.log

[the remark this answers is not in the document]
Comment (smuzaffar) [reply DataFormats/Common/test/testOneToManyAssociation.cc:71]: ah ok, this code is there to [check if it compiles](https://github.com/smuzaffar/cmssw/blob/890af695133ee08bc45431481193206ebb815fb0/DataFormats/Common/test/testOneToManyAssociation.cc#L30)

**Pull request merged**